src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java

```diff
@@ -68,13 +75,24 @@ static String stringOrNull(RequestParameter requestParameter) {
     return requestParameter == null ? null : requestParameter.getString();
   }
 
+  private static JsonObject jsonObjectRemoveNull(JsonObject obj) {
+    JsonObject n = new JsonObject();
+    for (String f : obj.fieldNames()) {
+      Object v = obj.getValue(f);
+      if (v != null) {
+        n.put(f, v);
+      }
+    }
```

Comment:
The obj.getValue(f) hash lookup should be avoided.

Removing the null value entries:
```
    obj.getMap().forEach((key, value) -> {
      if (value == null) {
        obj.remove(key);
      }
    });
```

For creating a copy:
```
    obj.getMap().forEach((key, value) -> {
      if (value != null) {
        n.put(key, value);
      }
    });
```

Reply:
UnsupportedOperationException: remove for all versions of in-place removal.

Reply:
Yes, should be
`obj.getMap().entrySet().removeIf(entry -> entry.getValue() == null);`